apps/analysis/tasks.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from contextlib import contextmanager
+
+from celery import shared_task
+from django.utils import timezone
+
+from apps.analysis.core import PipelineContext, StepContext
+from apps.analysis.log import LogEntry, Logger, LogLevel, LogStream
+from apps.analysis.models import AnalysisRun, RunStatus
+from apps.analysis.pipelines import get_data_pipeline, get_source_pipeline
+from apps.analysis.serializers import create_resource_for_data
+
+log = logging.getLogger(__name__)
+
+
+@contextmanager
+def run_context(run):
+    run.start_time = timezone.now()
+    run.status = RunStatus.RUNNING
+    run.save()
+
+    log_stream = RunLogStream(run)
+    llm_service = run.analysis.llm_provider.get_llm_service()
+    pipeline_context = PipelineContext(llm_service, logger=Logger(log_stream), params=run.params)
+
+    try:
+        yield pipeline_context
+        run.status = RunStatus.SUCCESS
+    except Exception as e:
+        run.status = RunStatus.ERROR
+        run.error = repr(e)
+        log.exception("Error running analysis")
```

Comment:
Suggestion: What if we `log.exception(e)` (so we see the exception details in Sentry) and rather set `run.error` to "Error running analysis"? I get that we'd want to tell the user what happened, but 1. would they understand exception messages? and 2. exposing exceptions doesn't sit quit right....or did you have some specific usecase in mind?

Reply:
`log.exception` will already capture the full exception details, no need to pass in the exception instance.

> I get that we'd want to tell the user what happened, but 1. would they understand exception messages? and 2. exposing exceptions doesn't sit quit right....or did you have some specific usecase in mind?

Absolutely. This is a temporary setup for now while we're still building.